monitoring/monitorlib/fetch/__init__.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+import json
+from typing import Dict, Optional
+
+import flask
+import requests
+import yaml
+from yaml.representer import Representer
+
+from monitoring.monitorlib import infrastructure
+
+
+def coerce(obj: Dict, desired_type: type):
+  if isinstance(obj, desired_type):
+    return obj
+  else:
+    return desired_type(obj)
+
+
+class RequestDescription(dict):
+  @property
+  def token(self) -> Dict:
+    return infrastructure.get_token_claims(self.get('headers', {}))
+yaml.add_representer(RequestDescription, Representer.represent_dict)
```

Comment:
is this line supposed to be indented?

Reply:
No; this line is telling the yaml renderer that the type just defined (RequestDescription) should be rendered as a standard dict -- it corresponds with the type definition for RequestDescription (directly prior).  Basically, we define the class, and then we register the render method of that class with the yaml library -- this line is the latter step.

Reply:
Cool, I've never done something like this before, good to know. Thanks